### PR TITLE
Protocol relative baseUri support

### PR DESCRIPTION
**Pull request by Ezaki113**
Hello, this pull request provide support for protocol relative baseUri, e.g. //some.host/api

### CODE REVIEW

**Comment by dice4x4** [comment]
👍

**Comment by pies** [comment]
@Ezaki113 Please change the `uri` back to `url` in all instances it was changed.

**Comment by pies** [comment]
@Ezaki113 Sorry, I'll try to sound more like a human being. Why did you change the name from `url` to `uri`? :)

File: src/ApiDefinition.php [hunk 1]
```diff
@@ -382,27 +386,19 @@ public function setVersion($version)
...
+        return ($this->version) ? str_replace('{version}', $this->version, $this->baseUri) : $this->baseUri;
     }
 
     /**
-     * Set the base url
-     *
-     * @param string $baseUrl
+     * @return string
      */
-    public function setBaseUrl($baseUrl)
```
Comment (Ezaki113) [inline src/ApiDefinition.php]: @brendo we really need setters in public interface?

**Comment by Ezaki113** [comment]
@pies because there are no baseUrl param in 0.8/1.0 specs and baseUri can also be uri template which is not URL

File: src/ApiDefinition.php (hunk 1 above)
Comment (pies) [inline src/ApiDefinition.php]: @Ezaki113 We do have this in the public interface. I don't think there's a good reason to remove it.

**Pull request merged**